Mage/src/main/java/mage/abilities/effects/common/ruleModifying/CombatDamageByToughnessEffect.java

```diff
@@ -19,12 +19,20 @@ public class CombatDamageByToughnessEffect extends ContinuousEffectImpl {
 
     private final FilterCreaturePermanent filter;
     private final boolean onlyControlled;
+    private final Duration duration;
 
     public CombatDamageByToughnessEffect(FilterCreaturePermanent filter, boolean onlyControlled) {
         super(Duration.WhileOnBattlefield, Layer.RulesEffects, SubLayer.NA, Outcome.Detriment);
         this.filter = filter;
         this.onlyControlled = onlyControlled;
     }
+     
+    public CombatDamageByToughnessEffect(FilterCreaturePermanent filter, boolean onlyControlled, Duration duration) {
+        super(this.duration, Layer.RulesEffects, SubLayer.NA, Outcome.Detriment);
+        this.filter = filter;
```

Comment:
remove this here, you can't access this before super is called and also you don't put this when access a function parameter.

Reply:
ok, how should i overwrite the duration then with the one from the constructor?

Reply:
i think i have an idea actually

Reply:
if you're gonna add a constructor for duration then only one should call the super